evm/src/cpu/kernel/interpreter.rs

```diff
@@ -537,6 +537,52 @@ impl<'a> Interpreter<'a> {
         self.push(U256::from_big_endian(hash.as_bytes()));
     }
 
+    fn run_callvalue(&mut self) {
+        self.push(
+            self.memory.context_memory[self.context].segments[Segment::ContextMetadata as usize]
+                .get(ContextMetadata::CallValue as usize),
+        )
+    }
+
+    fn run_calldataload(&mut self) {
+        let offset = self.pop().as_usize();
+        let value = U256::from_big_endian(
+            &(0..32)
+                .map(|i| {
+                    self.memory
+                        .mload_general(self.context, Segment::Calldata, offset + i)
+                        .byte(0)
+                })
+                .collect::<Vec<_>>(),
+        );
+        self.push(value);
+    }
+
+    fn run_calldatasize(&mut self) {
+        self.push(
+            self.memory.context_memory[self.context].segments[Segment::ContextMetadata as usize]
+                .get(ContextMetadata::CalldataSize as usize),
+        )
+    }
+
+    fn run_calldatacopy(&mut self) {
+        let dest_offset = self.pop().as_usize();
+        let offset = self.pop().as_usize();
+        let size = self.pop().as_usize();
+        for i in 0..size {
+            let calldata_byte = self
+                .memory
+                .mload_general(self.context, Segment::Calldata, offset + i)
+                .byte(0);
```

Comment:
nit: we could skip this we convert back to U256 below

Reply:
Good catch, done